src/neptune/new/internal/streams/std_stream_capture_logger.py

```diff
@@ -28,41 +29,51 @@ def __init__(self, container: MetadataContainer, attribute_name: str, stream: Te
         self.stream = stream
         self._thread_local = threading.local()
         self.enabled = True
+        self._log_data_queue = Queue()
+        self._finished_logging = threading.Event()
+        self._logging_thread = threading.Thread(target=self.__proces_logs, daemon=True)
+        self._logging_thread.start()
 
     def write(self, data: str):
-        if not hasattr(self._thread_local, "inside_write"):
-            self._thread_local.inside_write = False
-
         self.stream.write(data)
-        if self.enabled and not self._thread_local.inside_write:
-            try:
-                self._thread_local.inside_write = True
-                self._logger.log(data)
-            finally:
-                self._thread_local.inside_write = False
+        self._log_data_queue.put_nowait(data)
```

Comment:
Probably we should somehow handle `Full` exception (https://docs.python.org/3/library/queue.html#queue.Queue.put_nowait).
Maybe write some warning to stdout without logging it to Neptune?

Reply:
Should I add a size limit as well? Because currently the queue has an infinite size and can't be full.

Reply:
Oh, ok. Then let's keep it as it is.